### PR TITLE
Added a dark theme option for bed-time reading.

### PR DESCRIPTION
**Pull request by Caliandroid**
Issue  https://github.com/wallabag/android-app/issues/13
Night reading mode.

Remark:
Because of the API 8 compability, a restart is needed to take the dark theme in effect 
(recreate() is only available from API 11 on).

### CODE REVIEW

File: app/src/main/java/fr/gaulupeau/apps/Poche/Poche.java [hunk 1]
```diff
@@ -227,11 +212,13 @@ private void getSettings() {
...
         apiUsername = settings.getString("APIUsername", "");
         apiToken = settings.getString("APIToken", "");
+        nightmode = settings.getBoolean("Nightmode", false);
+
     }
 
     private void getDatabase() {
-	    if (database == null) {
-		    ArticlesSQLiteOpenHelper helper = new ArticlesSQLiteOpenHelper(this);
+        if (database == null || !database.isOpen()) {
```
Comment (Caliandroid) [inline app/src/main/java/fr/gaulupeau/apps/Poche/Poche.java]: this was the reason for the crash that appeared if the app was closed (main activity) and reopened.
The db connection was not null but closed.

File: app/src/main/java/fr/gaulupeau/apps/Poche/Poche.java [hunk 2]
```diff
@@ -256,18 +245,25 @@ protected void onPause() {
     @Override
     protected void onDestroy() {
         super.onDestroy();
-        if (database != null) {
+        if (database.isOpen()) {
```
Comment (Caliandroid) [inline app/src/main/java/fr/gaulupeau/apps/Poche/Poche.java]: Not sure if really necessary

File: app/src/main/java/fr/gaulupeau/apps/Poche/Poche.java [hunk 3]
```diff
@@ -256,18 +245,25 @@ protected void onPause() {
...
 
     private void updateUnread() {
         runOnUiThread(new Runnable() {
             public void run() {
                 ArticlesSQLiteOpenHelper helper = new ArticlesSQLiteOpenHelper(getApplicationContext());
-		getDatabase();
-                int news = database.query(ARTICLE_TABLE, null, ARCHIVE + "=0", null, null, null, null).getCount();
-                btnGetPost.setText(String.format(getString(R.string.btnGetPost), news));
+                getDatabase();
+                if (database.isOpen()) {
```
Comment (Caliandroid) [inline app/src/main/java/fr/gaulupeau/apps/Poche/Poche.java]: only called if db access is open. should be ok after the modification in getDatabase()

**Comment by tcitworld** [comment]
Can you rebase your changes against dev branch (there's many changes) and resubmit the PR on dev branch ? Thanks a lot !

**Comment by Caliandroid** [comment]
Yes i will try to rebase it.

**Comment by tcitworld** [comment]
Thanks a lot !

**Pull request closed without merging**